docs/extensions.rst

```diff
@@ -0,0 +1,43 @@
+Extensions
+==========
+
+The ASGI specification provides for server-specific extensions to be
+used outside of the core ASGI specification. This document specifies
+some common extensions.
+
+
+Websocket Denial Response
+-------------------------
+
+Websocket connections start with the user agent sending a HTTP request
+containing the appropriate upgrade headers. On receipt of this request
+a server can choose to either upgrade the connection or respond with a
+HTTP response (denying the upgrade). The core ASGI specification does
+not allow for any control over the denial-response, instead specifying
+that the HTTP status code ``403`` should be returned, whereas this
+extension allows an ASGI framework to control the
+denial-resposne. This is an extension, rather than a core part of
+ASGI, as most user agents do not utilise the denial response and hence
+this is a niche feature.
+
+ASGI Servers that implement this extension will provide
+``websocket.http.response.start`` and ``websocket.http.response.body``
```

Comment:
Let's have one key, not two. What would it mean if it just provided `websocket.http.response.body`? It wouldn't make sense. Also not a fan of that namespacing, maybe just have `websocket.something`.